unstable/wlr-screencopy-unstable-v1.xml

```diff
@@ -0,0 +1,141 @@
+
+    <event name="source_output">
+      <description summary="advertise an output source">
+        Advertise an output source available for capture.
+      </description>
+      <arg name="id" type="uint" summary="the source identifier"/>
+      <arg name="output" type="object" interface="wl_output"/>
+    </event>
+
+    <event name="source_remove">
+      <description summary="remove a source">
+        This event is sent whenever a source isn't available anymore.
+      </description>
+      <arg name="id" type="uint" summary="the source identifier"/>
+    </event>
+
+    <request name="capture">
+      <description summary="start capturing">
+        Request to start capturing from a source with a given identifier.
+      </description>
+      <arg name="frame" type="new_id" interface="zwlr_screencopy_frame_v1"/>
+      <arg name="id" type="uint" summary="the source identifier"/>
+    </request>
+
+    <request name="destroy" type="destructor">
+      <description summary="destroy the manager">
+        All objects created by the manager will still remain valid, until their
+        appropriate destroy request has been called.
+      </description>
+    </request>
+  </interface>
+
+  <interface name="zwlr_screencopy_frame_v1" version="1">
+    <description summary="a frame ready for copy">
+      This object represents a frame which is ready to have its resources
+      fetched and used.
+    </description>
+
+    <enum name="flags" bitfield="1">
+      <entry name="y_invert" value="1" summary="contents are y-inverted"/>
+    </enum>
+
+    <event name="buffer">
+      <description summary="buffer information">
+        Provides information about the frame's buffer. This event is sent once
+        as soon as the frame is created.
+
+        The client should then create a buffer with the provided width and
+        height, and send a copy request. It can optionally create a buffer with
+        the preferred format and stride.
+      </description>
+      <arg name="width" type="uint" summary="buffer width"/>
+      <arg name="height" type="uint" summary="buffer height"/>
+      <arg name="scale" type="uint" summary="scaling factor"/>
+      <arg name="transform" type="uint" summary="transformation, has the same
+        values as wl_output::transform"/>
+      <arg name="flags" type="uint" enum="flags" summary="buffer flags"/>
+      <arg name="format" type="uint" summary="preferred DRM_FORMAT"/>
+      <arg name="stride" type="uint" summary="preferred stride"/>
+    </event>
+
+    <request name="copy">
+      <description summary="copy the frame">
+        Copy the frame to the supplied buffer. The buffer must have a the
+        correct size, see zwlr_screencopy_frame_v1.buffer. The buffer needs to
+        have a supported format.
+
+        If the frame is successfully copied, a ready event is sent. Otherwise,
+        an abort event is sent.
+      </description>
+      <arg name="buffer" type="object" interface="wl_buffer"/>
```

Comment:
Not sure we want to use a `wl_buffer` for this - maybe a FD would be more appropriate.

Reply:
wl_buffer seems fine to me